src/background.js

```diff
@@ -86,3 +104,122 @@ if (isDevelopment) {
     })
   }
 }
+
+function main() {
+  axios.get(LATEST_RELEASES_URL).then(async result => {
+    const release = result.data.assets.find(
+      asset =>
+        asset.browser_download_url.includes(arch) && asset.browser_download_url.includes(platform)
+    )
+
+    if (release) {
+      const releaseUrl = release.browser_download_url
+      const latestReleaseVersion = releaseUrl
+        .match(/[/](\d+[.]){2}(\d+)[/]/)[0]
+        .replace('/', '')
+        .replace('/', '')
+      console.log('Latest release version: ' + latestReleaseVersion)
+      const isRunning = await isWalletRunning()
+
+      if (isRunning) {
+        const runningVersion = getCurrentWitnetVersion()
+        if (latestReleaseVersion !== runningVersion) {
+          // stop wallet process
+        }
+      } else {
+        if (!fs.existsSync(SHEIKAH_PATH)) {
+          fs.mkdirSync(SHEIKAH_PATH)
+        }
+
+        const existVersionFile = fs.existsSync(path.join(SHEIKAH_PATH, 'version.txt'))
+        const isLastestVersion =
+          latestReleaseVersion ===
+          fs.readFileSync(path.join(SHEIKAH_PATH, 'version.txt')).toString()
+
+        if (!existVersionFile || !isLastestVersion) {
+          await downloadWalletRelease(releaseUrl, latestReleaseVersion)
+        }
+        runWallet()
+      }
+    } else {
+      console.log('Your OS is not supported yet')
+    }
+  })
+}
+
+if (isDevelopment) {
+  main()
+}
+
+async function downloadWalletRelease(releaseUrl, version) {
+  return new Promise((resolve, reject) => {
+    axios
+      .get(releaseUrl, { responseType: 'stream' })
+      .then(async response => {
+        const file = `witnet-release-${arch}-${platform}.tar.gz`
+        const pipeline = util.promisify(stream.pipeline)
+        // Promise equivalent for response.data.pipe(writeStream)
+        await pipeline(response.data, fs.createWriteStream(file))
+        console.log('witnet release downloaded succesfully')
+
+        console.log('Decompressing release...')
+        // Decompress tar.gz file
+        tar.x({
+          file,
+          sync: true,
+        })
+        // TODO: tar is not decompressing correctly if a pass a path.
+        // Create these files in ./sheikah
+        fs.copyFileSync('witnet', `${SHEIKAH_PATH}/witnet`)
+        fs.copyFileSync('witnet.toml', `${SHEIKAH_PATH}/witnet.toml`)
+        fs.writeFileSync(path.join(SHEIKAH_PATH, 'version.txt'), version)
+
+        resolve()
+      })
+      .catch(err => {
+        console.log('err', err)
+      })
+  })
+}
+
+async function isWalletRunning() {
```

Comment:
Why not just ping the wallet using JSON-RPC? You know the socket address, that way you don't need to parse the full list of processes.